src/tests/UnitTests/IO/VersionOutputProcessorTests.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using FluentAssertions;
+using NUnit.Framework;
+using TestUtils;
+using GitHub.Unity;
+
+namespace UnitTests
+{
+    [TestFixture]
+    class VersionOutputProcessorTests : BaseOutputProcessorTests
+    {
+        public static IEnumerable<TestCaseData> ShouldParseVersionOutputs_TestCases()
+        {
+            TestCaseData testCase;
+
+            testCase = new TestCaseData(
+                $"git version 2.11.1.windows.1{Environment.NewLine}",
+                new Version(2, 11, 1));
+
+            testCase.SetName("Windows 2.11.1");
+            yield return testCase;
+        }
```

Comment:
This is where we should have lots and lots of weird-ass version strings to throw at the processor

Reply:
Does every git report 3 parts to the version followed by an OS string?

Reply:
I added the git version and git lfs version from my mac